src/equations/shallow_water_two_layer_1d.jl

```diff
@@ -0,0 +1,536 @@
+            - 0.09ω*sin(t + ω*x[1]) - 0.2ω*sin(2ω*x[1])) + 0.1cos(t + ω*x[1]) + 
+            0.1ω*cos(t + ω*x[1]) + 5.0(0.1ω*cos(t + ω*x[1]) + 0.2ω*sin(2ω*x[1]))*(2.0 + 0.2sin(t +
+            ω*x[1]) - 0.2cos(2ω*x[1])) + 0.2ω*sin(2ω*x[1]))
+
+    return SVector(du1, du2, du3, du4, 0.0)
+  end
+
+
+  """
+      boundary_condition_slip_wall(u_inner, orientation_or_normal, x, t, surface_flux_function,
+                                    equations::TwoLayerShallowWaterEquations1D)
+
+  Create a boundary state by reflecting the normal velocity component and keep
+  the tangential velocity component unchanged. The boundary water height is taken from
+  the internal value.
+
+  For details see Section 9.2.5 of the book:
+  - Eleuterio F. Toro (2001)
+    Shock-Capturing Methods for Free-Surface Shallow Flows
+    1st edition
+    ISBN 0471987662
+  """
+  @inline function boundary_condition_slip_wall(u_inner, orientation_or_normal, direction,
+      x, t, surface_flux_function, equations::TwoLayerShallowWaterEquations1D)
+    # create the "external" boundary solution state
+    u_boundary = SVector(u_inner[1],
+                        -u_inner[2],
+                         u_inner[3],
+                        -u_inner[4],
+                         u_inner[5])
+
+    # calculate the boundary flux
+    if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+      f = surface_flux_function(u_inner, u_boundary, orientation_or_normal, equations)
+    else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+      f = surface_flux_function(u_boundary, u_inner, orientation_or_normal, equations)
+    end
+    return f
+  end
+
+
+  # Calculate 1D flux for a single point
+  # Note, the bottom topography has no flux
+  @inline function flux(u, orientation::Integer, equations::TwoLayerShallowWaterEquations1D)
+    h1, h1_v1, h2, h2_v2, _ = u
+
+    # Calculate velocities
+    v1, v2 = velocity(u, equations)
+    # Calculate pressure
+    p1 = 0.5 * equations.gravity * h1^2
+    p2 = 0.5 * equations.gravity * h2^2
+
+    f1 = h1_v1
+    f2 = h1_v1 * v1 + p1
+    f3 = h2_v2
+    f4 = h2_v2 * v2 + p2
+
+    return SVector(f1, f2, f3, f4, zero(eltype(u)))
+  end
+
+
+  """
+      flux_nonconservative_wintermeyer_etal(u_ll, u_rr, orientation::Integer,
+                                            equations::TwoLayerShallowWaterEquations1D)
+
+  Non-symmetric two-point volume flux discretizing the nonconservative (source) term
+  that contains the gradient of the bottom topography [`TwoLayerShallowWaterEquations1D`](@ref).
+
+  Further details are available in the paper:
+  - Niklas Wintermeyer, Andrew R. Winters, Gregor J. Gassner and David A. Kopriva (2017)
+    An entropy stable nodal discontinuous Galerkin method for the two dimensional
+    shallow water equations on unstructured curvilinear meshes with discontinuous bathymetry
+    [DOI: 10.1016/j.jcp.2017.03.036](https://doi.org/10.1016/j.jcp.2017.03.036)
+  """
```

Comment:
We either need the final name before merging this (with an unfinished reference in the docstring) or a clear experimental warning, stating that this can change any time. Otherwise, we would be forced to make a breaking release when changing the name.

Reply:
Okay, I have added an experimental warning to the fluxes in af459aa39413652292357131cf0867f7a026f076